Cannot build for target `i686-pc-windows-msvc`

## Required information

**Operating system:** Windows 11 24H2 10.0.26100.712

**Rust version:** rustc 1.80.0-nightly (9cdfe285c 2024-05-22)

**Cargo version:** cargo 1.80.0-nightly (84dc5dc11 2024-05-20)

**iceoryx2 version:** `main` branch

**Observed result or behaviour:**
```pwsh
> cargo build --target i686-pc-windows-msvc
...
error: failed to run custom build command for `iceoryx2-pal-posix v0.3.0 (https://github.com/eclipse-iceoryx/iceoryx2.git?branch=main#1f5c8a3d)`

Caused by:
  process didn't exit successfully: `C:\test\target\debug\build\iceoryx2-pal-posix-bcebb1f44bc964b5\build-script-build` (exit code: 101)
  --- stdout
  cargo:rerun-if-changed=src/c/posix.h
  cargo:rerun-if-env-changed=TARGET
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS_i686-pc-windows-msvc
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS_i686_pc_windows_msvc
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS
  cargo:rerun-if-changed=src/c/posix.h

  --- stderr
  src/c/posix.h:32:10: fatal error: 'arpa/inet.h' file not found
  thread 'main' panicked at C:\Users\Chaoses\.cargo\git\checkouts\iceoryx2-1e1d4490bb91753f\1f5c8a3\iceoryx2-pal\posix\build.rs:44:18:
  Unable to generate bindings: ClangDiagnostic("src/c/posix.h:32:10: fatal error: 'arpa/inet.h' file not found\n")
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Obviously, `posix.h` only handles `_WIN64`, not `_WIN32`:
https://github.com/eclipse-iceoryx/iceoryx2/blob/1f5c8a3dbe6a4fdcb786a97e181b2b3637575fc7/iceoryx2-pal/posix/src/c/posix.h#L31-L46

Is 32-bit Windows not in the support list of iceoryx2?

**Expected result or behaviour:** Build successfully.

**Conditions where it occurred / Performed steps:** `i686-pc-windows-msvc` target

--- a/doc/release-notes/iceoryx2-unreleased.md
+++ b/doc/release-notes/iceoryx2-unreleased.md
@@ -54,7 +54,7 @@
 
  <!-- NOTE: Add new entries sorted by issue number to minimize the possibility of conflicts when merging. -->
 
- * Example text [#1](https://github.com/eclipse-iceoryx/iceoryx2/issues/1)
+ * Build failure for Windows 11 i686-pc-windows-msvc [#235](https://github.com/eclipse-iceoryx/iceoryx2/issues/235)
 
 ### Refactoring
 

--- a/iceoryx2-pal/posix/src/c/posix.h
+++ b/iceoryx2-pal/posix/src/c/posix.h
@@ -28,7 +28,7 @@
 #include <mqueue.h>
 #endif
 
-#ifndef _WIN64
+#if !(defined(_WIN64) || defined(_WIN32))
 #include <arpa/inet.h>
 #include <dirent.h>
 #include <grp.h>
@@ -102,10 +102,10 @@ char *acl_to_text(acl_t, ssize_t *) { return NULL; }
 acl_t acl_from_text(const char *) { return 0; }
 #endif
 
-#ifndef _WIN64
+#if !(defined(_WIN64) || defined(_WIN32))
 struct iox2_sigaction {
-    size_t iox2_sa_handler;
-    sigset_t iox2_sa_mask;
-    int iox2_sa_flags;
+  size_t iox2_sa_handler;
+  sigset_t iox2_sa_mask;
+  int iox2_sa_flags;
 };
 #endif

--- a/iceoryx2-pal/posix/src/c/scandir.c
+++ b/iceoryx2-pal/posix/src/c/scandir.c
@@ -10,7 +10,7 @@
 //
 // SPDX-License-Identifier: Apache-2.0 OR MIT
 
-#ifndef _WIN64
+#if !(defined(_WIN64) || defined(_WIN32))
 #include <dirent.h>
 
 int scandir_ext(const char *dir, struct dirent ***namelist) {

--- a/iceoryx2-pal/posix/src/c/sigaction.c
+++ b/iceoryx2-pal/posix/src/c/sigaction.c
@@ -10,7 +10,7 @@
 //
 // SPDX-License-Identifier: Apache-2.0 OR MIT
 
-#ifndef _WIN64
+#if !(defined(_WIN64) || defined(_WIN32))
 #include "posix.h"
 
 int iox2_sigaction_func(int sig, const struct iox2_sigaction *restrict act,

--- a/iceoryx2-pal/posix/src/c/socket_macros.c
+++ b/iceoryx2-pal/posix/src/c/socket_macros.c
@@ -10,7 +10,7 @@
 //
 // SPDX-License-Identifier: Apache-2.0 OR MIT
 
-#ifdef _WIN64
+#if defined(_WIN64) || defined(_WIN32)
 // clang-format off
 // the include order is important, since some headers are defining macros that
 // are used in the next header

--- a/iceoryx2-pal/posix/src/windows/mman.rs
+++ b/iceoryx2-pal/posix/src/windows/mman.rs
@@ -37,7 +37,7 @@ use windows_sys::Win32::{
 
 use super::win32_handle_translator::{FdHandleEntry, FileHandle, HandleTranslator, ShmHandle};
 
-const MAX_SUPPORTED_SHM_SIZE: usize = 1024 * 1024 * 1024;
+const MAX_SUPPORTED_SHM_SIZE: u64 = 1024 * 1024 * 1024;
 
 pub unsafe fn mlock(addr: *const void, len: size_t) -> int {
     -1

